engine/src/main/java/org/n52/javaps/engine/impl/EngineImpl.java

```diff
@@ -182,20 +192,25 @@ public JobId execute(OwsCode identifier,
 
     private Result onJobCompletion(Job job) throws EngineException {
         try {
+            this.jobStatusLock.writeLock().lock();
```

Comment:
you should lock it before the `try`

Reply:
I was wondering, this is just for performance reasons, right?